### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by harshi0102**
### Unfortunately, my coding partner who is assigned to me this week was not collaborating with me properly, so after speaking to Student success team of microverse, I am solo this week for now.
The Student Success team has given me permission to continue this week solo. If you have any questions please contact Marco Lizza via the Code Reviewers Slack workspace.
Please find the screenshot below:
image

image
Blog app: add devise project done according to the requirements which are:

Remove current_user method in ApplicationController, devise will provide us with one.
[Install and setup devise](https://github.com/heartcombo/devise#getting-started).
Can register a new user.
User logs in with a combination of email and password.
Hashed password should be stored in the database.
Ask for confirmation of email.
Can reset password.
[Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,11 +1,16 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (joel-kalema) [inline app/models/user.rb]: - [ ] You did a good job adding devise 🥇, however, make sure the `:confirmable` is enabled in the user model 👍. this is to make the application asks for confirmation of email. It'll be something like this:

```ruby
devise :database_authenticatable, :registerable,
		:recoverable, :rememberable, :validatable, :confirmable
``` 

Kindly, fix it 😃

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @harshi0102 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by harshi0102** [comment]
**According to the previous feedback  , I added confirmable to the user model -** 
https://github.com/harshi0102/blog-app-rails/blob/969e80fc6eed08aaf8118d8ebad37f4655f14440/app/models/user.rb#L5

**The confirmation token needs to be clicked before signing to the page. After verifiying I have a screenshot below which generates confirmation token for login to the page successfully.**

image

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  <div class="field">
+      <%= f.text_field :name, autofocus: true,  placeholder: "name",autocomplete: "name" %>
+  </div>
+
+  <div class="field">
+    <%= f.email_field :email, autofocus: true,  placeholder: "email",autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+      <%= f.text_area :bio, autofocus: true,  placeholder: "bio",autocomplete: "bio" %>
```
Comment (Munsa1) [inline app/views/devise/registrations/new.html.erb:15]: - [ ] (optional) Consider matching the given wireframe for your Sign up and Login page so as to fully meet the given requirements

Your design
image

Given wireframe
image

Check project [requirement bullet point number three](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md)
image

**Review by Munsa1 — APPROVED** [review]
Hi @harshi0102, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Change By Previous Reviewer :heavy_check_mark: 

## Optional suggestions
- [ ] Check inline comments

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**